client/a8c-for-agencies/components/items-dashboard/site-favicon/index.tsx

```diff
@@ -18,14 +23,24 @@ const SiteFavicon = ( {
 	isDotcomSite = false,
 	size = 40,
 	className = '',
+	fallback = 'color',
 }: SiteFaviconProps ) => {
+	const { __ } = useI18n();
 	const siteColor = color ?? 'linear-gradient(45deg, #ff0056, #ff8a78, #57b7ff, #9c00d4)';
+	const site = useSelector( ( state ) => getSite( state, blogId ) );
 
-	const defaultFavicon = isDotcomSite ? (
-		<WordPressLogo className="wpcom-favicon" size={ size * 0.8 } />
-	) : (
-		<div className="no-favicon" style={ { background: siteColor } } />
-	);
+	let defaultFavicon;
+	if ( isDotcomSite || fallback === 'wordpress-logo' ) {
+		defaultFavicon = <WordPressLogo className="wpcom-favicon" size={ size * 0.8 } />;
+	} else if ( fallback === 'color' ) {
+		defaultFavicon = <div className="no-favicon" style={ { background: siteColor } } />;
+	} else if ( fallback === 'first-grapheme' ) {
+		defaultFavicon = (
+			<div role="img" aria-label={ __( 'Site Icon' ) }>
+				{ getFirstGrapheme( site?.title ?? '' ) }
+			</div>
+		);
+	}
```

Comment:
This is a bit confusing for some reason. And it looks like we could have some sites with no favicon at all. The defaultFavicon is populated by an if that in theory could not set a favicon.

Reply:
> This is a bit confusing for some reason

Can you expand on bit on this? What are you finding confusing?

> it looks like we could have some sites with no favicon at all

Yeah, that's what we're handling here: the icon we want to show on sites without icon.

> The defaultFavicon is populated by an if that in theory could not set a favicon.

Not really because `fallback` can only have 3 values and we are handling them all here.

I could do this instead:

```
if ( isDotcomSite || fallback === 'wordpress-logo' ) {
 // ..
} else if ( fallback === 'color' ) {
 // ..
} else {
 // ..
}
```

But I thought it was clearer to include an explicit `fallback === 'first-grapheme'` check at the end.

Reply:
> Yeah, that's what we're handling here: the icon we want to show on sites without icon.

I mean, your code in its current state there exists a case where we won't return an icon.

It isn't obvious that your if handles all the possibilities (it actually isn't since there is no default case)

Reply:
> your code in its current state there exists a case where we won't return an icon.

No, because `fallback` is typed to be an enum of 3 values and we check all the three values.

Also, in the worst case scenario that we change `fallback` to have more values and we forget to add a new if, it won't be a problem because [an undefined `defaultFavicon` is also ok](https://github.com/Automattic/wp-calypso/blob/7b648ac24dc9fc084f28cf07eaab6b22dae9ff3f/client/blocks/site-icon/index.tsx#L64).

But anyway, I finally refactored the code in https://github.com/Automattic/wp-calypso/pull/90585/commits/c8bee3b3a49f245026b2c86a97979fd1f0000a2e and moved the `isDotcomSite` exception outside which is was preventing me to use a `switch` in the first place. I think all possibilities are explicitly handled now.

Reply:
Thanks for making the check exhaustive! ❤️